src/spi_bus.cpp

```diff
@@ -0,0 +1,364 @@
+        MotorDriver& udriver = main_board_interface_->motor_drivers[i];
+        udriver.motor1->SetCurrentReference(0);
+        udriver.motor2->SetCurrentReference(0);
+        udriver.motor1->Disable();
+        udriver.motor2->Disable();
+        udriver.DisablePositionRolloverError();
+        udriver.Disable();
+    }
+}
+
+/**
+ * Output and status
+ */
+
+std::shared_ptr<const MotorBoardInterface::ScalarTimeseries>
+SpiBus::get_measurement(
+    const size_t udriver_id,
+    const MotorBoardInterface::MeasurementIndex& index) const
+{
+    return measurement_[udriver_id * 2 + index];
+}
+
+std::shared_ptr<const MotorBoardInterface::StatusTimeseries>
+SpiBus::get_status(const size_t udriver_id) const
+{
+    return status_[udriver_id];
+}
+
+/**
+ * input logs
+ */
+
+std::shared_ptr<const MotorBoardInterface::ScalarTimeseries>
+SpiBus::get_control(const size_t udriver_id,
+    const MotorBoardInterface::ControlIndex& index) const
+{
+    return control_[udriver_id * 2 + index];
+}
+
+std::shared_ptr<const MotorBoardInterface::CommandTimeseries>
+SpiBus::get_command(const size_t udriver_id) const
+{
+    return command_[udriver_id];
+}
+
+std::shared_ptr<const MotorBoardInterface::ScalarTimeseries>
+SpiBus::get_sent_control(const size_t udriver_id,
+    const MotorBoardInterface::ControlIndex& index) const
+{
+    return control_[udriver_id * 2 + index];
+}
+
+std::shared_ptr<const MotorBoardInterface::CommandTimeseries>
+SpiBus::get_sent_command(const size_t udriver_id) const
+{
+    return sent_command_[udriver_id];
+}
+
+/**
+ * Setters
+ */
+
+void SpiBus::set_control(const size_t udriver_id, const double& control,
+                         const MotorBoardInterface::ControlIndex& index)
+{
+    control_[udriver_id * MotorBoardInterface::ControlIndex::control_count +
+             index]->append(control);
+}
+    
+void SpiBus::set_command(const size_t udriver_id, 
+                         const MotorBoardCommand& command)
+{
+    command_[udriver_id]->append(command);
+}
+
+void SpiBus::send_if_input_changed()
+{
+    // send command if a new one has been set ----------------------------------
+    bool commands_have_changed = false;
+    for(size_t i = 0 ; i < nb_udrivers_ ; ++i)
+    {
+        if(command_[i]->has_changed_since_tag())
+        {
+            commands_have_changed = true;
+            break;
+        }
+    }
+    if(commands_have_changed)
+    {
+        send_newest_command();
+    }
+    // send controls if a new one has been set ---------------------------------
+    bool controls_have_changed = false;
+    for(size_t i = 0 ; i < control_.size() ; ++i)
+    {
+        if(control_[i]->has_changed_since_tag())
+        {
+            controls_have_changed = true;
+            break;
+        }
+    }
+    if(controls_have_changed)
+    {
+        send_newest_controls();
+    }
+}
+
+void SpiBus::send_newest_command()
+{
+    for(size_t i = 0 ; i < nb_udrivers_ ; ++i)
+    {
+        // Some aliases
```

Comment:
you have to tag the command timeseries somewhere in here, otherwise it will just send every time